google/datalab/bigquery/commands/_bigquery.py

```diff
@@ -250,7 +250,26 @@ def _create_load_subparser(parser):
   return load_parser
 
 
-def _get_query_argument(args, cell, env):
+def _default_context_for_args(args):
+  """Get the default Context for the parsed arguments.
+
+  Args:
+    args: the dictionary of magic arguments.
+  Returns:
+    A Context based on the current default context, but with any explicitly specified
+      arguments overriding the defaults.
+  """
+  context = google.datalab.Context.default()
+  dialect_arg = args.get('dialect', None)
+  billing_tier_arg = args.get('billing', None)
+  if dialect_arg:
+    context.config['bigquery_dialect'] = dialect_arg
```

Comment:
doesn't this overwrite the global context for all future executions too? i think we need to modify the `.default()` method to make it return a new copy on every call, and add a `.global_context()` instead of it.